### ISSUE TITLE
Incompatible `LoadLatestOptions()` API change

### ISSUE THREAD

**Issue by ajkr** [facebook/rocksdb#7614]
Regression test is here: https://github.com/ajkr/rocksdb/tree/incompat-load-latest-options

Output:

```
$ ./options_util_test --gtest_filter=OptionsUtilTest.BadLatestOptions
Note: Google Test filter = OptionsUtilTest.BadLatestOptions
[==========] Running 1 test from 1 test case.
[----------] Global test environment set-up.
[----------] 1 test from OptionsUtilTest
[ RUN      ] OptionsUtilTest.BadLatestOptions
utilities/options/options_util_test.cc:579: Failure
Value of: s.IsInvalidArgument()
  Actual: false
Expected: true
[  FAILED  ] OptionsUtilTest.BadLatestOptions (1 ms)
[----------] 1 test from OptionsUtilTest (1 ms total)

[----------] Global test environment tear-down
[==========] 1 test from 1 test case ran. (1 ms total)
[  PASSED  ] 0 tests.
[  FAILED  ] 1 test, listed below:
[  FAILED  ] OptionsUtilTest.BadLatestOptions

 1 FAILED TEST
```

Here's a backport of the test on top of the 6.13 release branch, which shows this `LoadLatestOptions()` scenario used to return `InvalidArgument`: https://github.com/ajkr/rocksdb/tree/6.13.incompat-load-latest-options

**Comment by mrambacher** [comment]
Fixed.

**Comment by ajkr** [comment]
Sorry I should have included in the description, this is specifically about the "TableOptions/BlockBasedTable" section and the provided repro included the fix from #7612. Did you mean it is fixed for "TableOptions/BlockBasedTable" somewhere too?

**Comment by ajkr** [comment]
Fixed by #7615 (thanks!)
